.github/workflows/pathogen-repo-build.yaml

```
@@ -230,41 +212,19 @@ jobs:
           | "$NEXTSTRAIN_GITHUB_DIR"/bin/json-to-envvars
           | tee -a "$GITHUB_ENV"
 
-      - name: Set build runtime environment variables
-        env:
-          env: ${{ inputs.env || '[]' }}
-          secrets: ${{ toJson(secrets) || '' }}
-        run: |
-          # shellcheck disable=SC2154
-
-          envvars=$(echo "$env" | yq 'keys | join(" ")')
-          secrets=$(echo "$secrets" | jq -r 'del(.github_token) | keys | join(" ")')
-
-          # read exits with 1 when it encounters EOF
-          read -a varnames -r -d $'\0' <<<"$envvars $secrets" || true
-          "$NEXTSTRAIN_GITHUB_DIR"/bin/write-envdir "$ENV_DIR" "${varnames[@]}"
-
       - name: Run build via ${{ inputs.runtime }}
         env:
-          cpus: ${{ inputs.cpus }}
-          memory: ${{ inputs.memory }}
-          detach: ${{ inputs.runtime == 'aws-batch' || null }}
-          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
-          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          NEXTSTRAIN_BUILD_COMMAND: ${{ inputs.run }}
         run: |
           # shellcheck disable=SC2154
           set -x
 
-          nextstrain build \
-            --cpus "$cpus" \
-            ${memory:+--memory="$memory"} \
-            ${detach:+--detach} \
-            --exec env \
-            . \
-              envdir "$ENV_DIR" snakemake \
-              --cores "$cpus" \
-              ${{ inputs.build-args }} \
-          | tee build.log
+          if [[ "$NEXTSTRAIN_BUILD_COMMAND" == "nextstrain build "* ]]; then
+            eval "$NEXTSTRAIN_BUILD_COMMAND" | tee build.log
+          else
+            echo "::error::The run input must be a `nextstrain build` command"
+            exit 1
+          fi
```

@joverlee521 [wrote elsewhere](https://github.com/nextstrain/.github/pull/44#discussion_r1261718189) about this bit of code:

> Updated in [latest push](https://github.com/nextstrain/.github/compare/3f05452c6894e9b3f9ff50fb4c94c10650f5433f...b79ec646e73f496a7639e609de5365993bb3ae54?expand=1). I would love to know if there's a better way to run the provided command besides using `eval`.

Instead of `eval` you could directly interpolate at the workflow level (i.e. pre-execution):

```suggestion
          ${{ inputs.run }} |& tee build.log
```

The indirection of defining `NEXTSTRAIN_BUILD_COMMAND` isn't needed for input robustness since we're running it as a bit of shell anyway. I'd also not worry about if the input starts with `nextstrain build` or not. Maybe emitting a warning is useful? but wouldn't enforce a restriction.

Would also direct stderr to the build log (`|&`), not just stdout (`|`).

---

Ah, that makes sense. Updated in [force-push](https://github.com/nextstrain/.github/compare/1097fc9052d2a07c393a210aa57f1da09b6c1867..822187aa78c560d85be378490d3c666cf21457e5).

---

Err, slight hiccup with direct interpolation of `${{ inputs.run }}`. This retains newlines at the end of the command resulting in a [syntax error in a test seasonal-flu run](https://github.com/nextstrain/seasonal-flu/actions/runs/5604334827/jobs/10252159727#step:7:326). I thought changing the run block style to `>` in https://github.com/nextstrain/.github/pull/44/commits/357a79bd18274cb2ce2c929a5424576c3da733de would replace the newline with a space, but that [did not work](https://github.com/nextstrain/seasonal-flu/actions/runs/5604334827/jobs/10252608976#step:7:325).

Going back to eval again in https://github.com/nextstrain/.github/pull/44/commits/8440ef60e51b924a761ae52cbf0654f669267cd7 and this worked as expected for the [test seasonal-flu run](https://github.com/nextstrain/seasonal-flu/actions/runs/5604334827).

---

Ah, yeah. Using the `>` block style won't work because GitHub's `${{ … }}` interpolation is evaluated _after_ parsing the YAML (an ordering which is quite important). An alternative that I think would work is to setup the redirection before hand rather than as a pipeline, e.g.:

```bash
exec &> >(tee build.log)
${{ inputs.run }}
```

But I think also fine to keep `eval`.

---

Alternatively alternatively, you could also encase the thing in a subshell:

```bash
( ${{ inputs.build.log }} ) | tee build.log
```

or group command:

```bash
{ ${{ inputs.build.log }} ; } | tee build.log
```

---

(But really fine to keep `eval`.)